### PR TITLE
Add support for pool-specific ammo pips decoration

### LINKED ISSUE #19235: There is no way to tell a 'WithAmmoPipsDecoration' which 'AmmoPool' it belongs to
This poses a issue on units with multiple different Ammopools.
I have a plane that does 4 strafes of machinegun and carries 2 depthcharges to be used against submarines.
Previously was able to tell on the AmmoPool trait to display 2 seperate blue pip for the depthcharges, now after upgrading to playtest it no longer can be done.

### PR DESCRIPTION
**Pull request by reaperrr**
Closes #19235.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Render/WithAmmoPipsDecoration.cs [hunk 1]
```diff
@@ -49,7 +52,9 @@ public class WithAmmoPipsDecoration : WithDecorationBase<WithAmmoPipsDecorationI
 		public WithAmmoPipsDecoration(Actor self, WithAmmoPipsDecorationInfo info)
 			: base(self, info)
 		{
-			ammo = self.TraitsImplementing<AmmoPool>().ToArray();
+			ammo = !info.AmmoPools.Any() ? self.TraitsImplementing<AmmoPool>().ToArray() :
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/WithAmmoPipsDecoration.cs]: I personally would have written this as
```csharp
if (info.AmmoPools.Any())
	ammo = self.TraitsImplementing<AmmoPool>()
		.Where(ap => info.AmmoPools.Contains(ap.Info.Name))
		.ToArray();
else
	ammo = self.TraitsImplementing<AmmoPool>().ToArray();
```
which I feel is slightly more readable. Not a blocker though.

**Review by pchote — DISMISSED** [review]

**Comment by pchote** [comment]
IMO we should pick this to prep before the release to avoid an otherwise unnecessary modding regression.

**Comment by reaperrr** [comment]
Updated.

**Review by MlemandPurrs — APPROVED** [review]
Working as intended +1 from me.

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/7632a8d4a82dca31c6d49ca7a23e42a6c9693aa9...35d9fa15991d304a19e59067a880e8850dd1152a

https://github.com/OpenRA/OpenRA/commit/ab3f3000fd786901e7db9fe1337f0c9d643ad188

**Pull request merged**